### ISSUE TITLE
Throw clearer exception for np.diff on 0-dimensional ndarray

### ISSUE THREAD

**Issue by navneet-nmk** [numpy/numpy#13296]
np.diff throws axis exception for 0-dimensional arrays. Is this the default behaviour? Shouldn't it just return the array as it is?

### Reproducing code example:

```python
import numpy as np
foo = np.array(1.11111111111, np.float64)
print(foo.ndim)
print(np.diff(foo))
```

### Error message:

`Traceback (most recent call last):
  File "/Users/navneetmadhukumar/Downloads/pytorch-maml-rl/main.py", line 212, in <module>
    print(np.diff(foo))
  File "/Users/navneetmadhukumar/anaconda3/envs/rllab3/lib/python3.5/site-packages/numpy/lib/function_base.py", line 1241, in diff
    axis = normalize_axis_index(axis, nd)
numpy.AxisError: axis -1 is out of bounds for array of dimension 0`

### Numpy/Python version information:

1.16.0 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:52:12) 
[GCC 4.2.1 Compatible Apple LLVM 4.2 (clang-425.0.28)]

**Comment by mhvk** [comment]
It has to calculate differences, so needs more than 1 element, which suggests the error is correct. That said, for a 1-element one-dimensional array it returns an empty array. I'm not sure that would make more sense for a scalar input, though.

**Comment by mhvk** [comment]
@navneet-nmk - Essentially, `np.diff` assumes the array is iterable, which a scalar is not. I don't think we should change the behaviour, but since it was surprising to you, the documentation perhaps can be improved. What would have helped you?

**Comment by navneet-nmk** [comment]
I am actually developing for another repository a variant of diff and we have test cases with 0-dimensional arrays. I guess it helps to know that 0-dimensional arrays are not supported.

**Comment by mattip** [comment]
Perhaps the error message could be improved to `TypeError('diff requires more than one element')`.

**Comment by mhvk** [comment]
Yes, a `TypeError` would be better and is easy to do right after [getting the dimension of the input](https://github.com/numpy/numpy/blob/master/numpy/lib/function_base.py#L1237). I changed the title to reflect this.

**Comment by mattip** [comment]
Closing, fixed by #13298
